### PR TITLE
Changes the SingularitySystem to use generic visualizers

### PR DESCRIPTION
**Pull request by ZoldorfTheWizard**
## About the PR

The singularity had a lot of hardcoded rsi and old appearance stuff. this PR moves singularity to use a generic visualizer and updates some fields for better readability. also allows custom singularity adminbus stuff via prototypes :o)

**Media**

this PR does not require an ingame showcase

**Changelog**

### CODE REVIEW

File: Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml [hunk 1]
```diff
@@ -50,3 +53,33 @@
...
+    sprite: Structures/Power/Generation/Singularity/singularity_1.rsi
+    layers:
+    - map: [ "VisualLevel" ]
+      state: singularity_1
+  - type: GenericVisualizer
+    visuals:
+      enum.SingularityAppearanceKeys.Singularity:
+       VisualLevel:
+          1:
+            sprite: Structures/Power/Generation/Singularity/singularity_1.rsi
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml:66]: You dont have to specify the sprite each time

Comment (ZoldorfTheWizard) [reply Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml:66]: there is one sprite for each power level, how would i not change the sprite?

Comment (ElectroJr) [reply Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml:66]: yeah, in this case it does need to specify the sprite every time

File: Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml [hunk 2]
```diff
@@ -50,3 +53,33 @@
...
+          3:
+            sprite: Structures/Power/Generation/Singularity/singularity_3.rsi
+            state: singularity_3
+          4:
+            sprite: Structures/Power/Generation/Singularity/singularity_4.rsi
+            state: singularity_4
+          5:
+            sprite: Structures/Power/Generation/Singularity/singularity_5.rsi
+            state: singularity_5
+            scale: 1.5,1.5
```
Comment (ElectroJr) [inline Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml:80]: if you change the scale at level 5 & 6, you need to make sure to set it back to 1,1 for the other levels.

File: Content.Shared/Singularity/Components/SingularityComponent.cs [hunk 3]
```diff
@@ -87,14 +87,16 @@ public sealed class SingularityComponent : Component
...
     /// <summary>
-    /// The base sprite file and state of the singularity.
+    /// The base rsi key to use for updating the sprite layer.
+    /// 'singularity' if singularity_level or 'brodylarity' if using a custom singulo rsi like 'brodylarity_level.
     /// </summary>
-    [DataField("baseSprite")]
-    public SpriteSpecifier.Rsi BaseSprite = new SpriteSpecifier.Rsi(new ResourcePath("Structures/Power/Generation/Singularity/singularity"), "singularity");
+    ///
+    [DataField("spriteBaseRsi")]
+    public string SpriteBaseRsi = "singularity";
```
Comment (ElectroJr) [inline Content.Shared/Singularity/Components/SingularityComponent.cs:99]: These shouldn't be required anymore

File: Content.Client/Singularity/Systems/SingularitySystem.cs [hunk 4]
```diff
@@ -45,7 +45,7 @@ protected override void OnSingularityStartup(EntityUid uid, SingularityComponent
         base.OnSingularityStartup(uid, comp, args);
         if (TryComp<SpriteComponent>(uid, out var sprite))
         {
-            sprite.LayerMapReserveBlank(comp.Layer);
+            sprite.LayerMapReserveBlank(comp.SpriteLayerKey);
         }
     }
```
Comment (ElectroJr) [inline Content.Client/Singularity/Systems/SingularitySystem.cs:50]: You should be able to just remove `OnSingularityStartup` and `OnAppearanceChange` altogether

File: Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml [hunk 5]
```diff
@@ -50,3 +53,33 @@
   - type: WarpPoint
     follow: true
     location: singularity
+  - type: Sprite
```
Comment (ElectroJr) [inline Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml:56]: sprite is already defined a bit further up, you need to merge them

File: Content.Shared/Singularity/Components/SingularityComponent.cs (hunk 3 above)
Comment (ZoldorfTheWizard) [reply Content.Shared/Singularity/Components/SingularityComponent.cs]: pure magic

File: Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml (hunk 5 above)
Comment (ZoldorfTheWizard) [reply Resources/Prototypes/Entities/Structures/Power/Generation/Singularity/singularity.yml:49]: fixed

File: Content.Client/Singularity/Systems/SingularitySystem.cs (hunk 4 above)
Comment (ZoldorfTheWizard) [reply Content.Client/Singularity/Systems/SingularitySystem.cs]: fucking magic

**Pull request merged**